Support searching by commit for results.webkit.org

facetothefate opened the pull request:
#### 67030dad6ff1308ad9512395a23f3a69d92375d9
<pre>
Support searching by commit for results.webkit.org
<a href="https://bugs.webkit.org/show_bug.cgi?id=256809">https://bugs.webkit.org/show_bug.cgi?id=256809</a>

Reviewed by Jonathan Bedard.

* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/timeline.js: Add commit search input and jump to next regess point button
* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/library/css/webkit.css: Add new highlight color and make timeline series seamless
* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/library/js/components/TimelineComponents.js: Add new search API, highlight and jump to highlighted scale, add new search dot and hightlight the dot&apos;s scale API

Canonical link: <a href="https://commits.webkit.org/270406@main">https://commits.webkit.org/270406@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ad3697d1133368b51d4f57499d92174182f1785f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25419 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4023 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/26704 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27533 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/23315 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/25695 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/5752 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1460 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/23484 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/25664 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/2972 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/21935 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28111 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/2635 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/22879 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/28998 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/22120 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/23199 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/23243 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/26819 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/24653 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/2619 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/873 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/32083 "Built successfully") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/3990 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/6086 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3069 "Built successfully") | | [⏳ 🛠 jsc-mips ](https://ews-build.webkit.org/#/builders/JSC-MIPSEL-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3246 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/2958 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [⏳ 🧪 jsc-mips-tests ](https://ews-build.webkit.org/#/builders/JSC-MIPSEL-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/library/js/components/TimelineComponents.js:
```diff
@@ -410,12 +423,18 @@ Timeline.CanvasSeriesComponent = (dots, scales, option = {}) => {
-    const height = option.height ? option.height : 2 * radius + tagHeight + selectedOutLineSize;
+    let height = option.height ? option.height : 2 * radius + tagHeight + selectedOutLineSize;
+    const headerHeight = parseInt(computedStyle.getPropertyValue('--mediumSize')) + 8;
+    // The header lable's height is mediumSize plus 8 margin
```
I don't think we need this comment.

JonWBedard commented on Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/timeline.js:
```diff
@@ -1013,21 +1021,52 @@ class TimelineFromEndpoint {
+        });
+        return `<div> 
+            <div class="input">
+                <input type="text" ref="${searchInputRef}" autocomplete="off" autocapitalize="none" required/>
```
I think overloading `cntrl-f` would be better than a dedicated input field...although that might be quite a bit of work

JonWBedard reviewed (commented):
Note typo in the first line of the commit message.

bls1999 commented on Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/css/search.css:
```diff
@@ -45,4 +45,18 @@
     color: white;
+}
+
+.next-regress-bar {
```
Can we add some padding on top here?

JonWBedard replied:
Agree on the padding, the top of the button ends up hidden when scrolling at the moment.

bls1999 commented on Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/css/search.css:
```diff
@@ -45,4 +45,18 @@
+
+@media (prefers-color-scheme: dark) {
+    .next-regress-bar {
+        background-color: var(--blurBackgroundColorDark);
```
This background color doesn't match the section's background color (should be `grayDarker`, is `blurBackgroundColorDark`).

facetothefate replied:
There is a reason for that, when you have a lot of timelines, this search bar can hover on the top. so you don't need to scroll back and hit the button. it will be better to give this a different color to inform user this is a float bar.

JonWBedard replied:
Why does this have a background at all? With only a single button, I think I'd rather have this be transparent.

bls1999 reviewed (changes requested):
One thing (not mentioned in the specific comments): the jumping behavior seems to be random when there are flaky failures on different platforms.

These changes would be a HUGE win for the team. Thank you so much for this PR!

JonWBedard commented on Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/timeline.js:
```diff
@@ -1026,21 +1063,108 @@ class TimelineFromEndpoint {
+        return `<div ref="${containnerRef}" style="position: relative">
+            <div ref="${searchBarRef}" class="next-regress-bar">
+                <button class="button" ref="${nextRegressButtonRef}">
+                    Next Regress Point
```
I feel like we should have a `Prev Regress Point` button too.

JonWBedard commented on Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/timeline.js:
```diff
@@ -1026,21 +1063,108 @@ class TimelineFromEndpoint {
+                    this.selectedDotsButtonGroupRef.setState({show: false});
+                },
+                onSelect: (dots, selectedDotRect, seriesRect, e) => {
+                    // this api will called with selected dots for each series once, and compose the selectedDotRect during the call
```
I don't know that this comment adds anything, especially when duplicated 3 lines down.

JonWBedard commented on Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/timeline.js:
```diff
@@ -1026,21 +1068,195 @@ class TimelineFromEndpoint {
-            customizedLayer: `<div style="position:absolute" ref="${this.selectedDotsButtonGroupRef}"></div>`,
-            onSelecting: (e) => {
-                this.selectedDotsButtonGroupRef.setState({show: false});
+
```
Nit: Should only have a single newline.

JonWBedard reviewed (approved):
Minor style nit, otherwise good to go.

facetothefate committed:
Support searching by commit for results.webkit.org
https://bugs.webkit.org/show_bug.cgi?id=256809

Reviewed by Jonathan Bedard.

* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/timeline.js: Add commit search input and jump to next regess point button
* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/library/css/webkit.css: Add new highlight color and make timeline series seamless
* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/library/js/components/TimelineComponents.js: Add new search API, highlight and jump to highlighted scale, add new search dot and hightlight the dot's scale API

Canonical link: https://commits.webkit.org/270406@main
